crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -401,205 +268,4 @@ mod tests {
-            assert_eq!(1000, total_pruned.unwrap());
-        }
-        {
-            // add 3 versions for every object id and 1000 unique object ids
-            let total_pruned = test_pruning(
-                tempfile::tempdir()?.into_path(),
-                /* num_versions_per_object */ 2,
-                /* num_object_versions_to_retain */ 2,
-                /* total_unique_object_ids */ 1000,
-            )
-            .await;
-            // We had only 2 version per key, we did not prune anything
-            assert_eq!(0, total_pruned.unwrap());
-        }
-        {
-            // add 3 versions for every object id and 1000 unique object ids
-            let total_pruned = test_pruning(
-                tempfile::tempdir()?.into_path(),
-                /* num_versions_per_object */ 1,
-                /* num_object_versions_to_retain */ 2,
-                /* total_unique_object_ids */ 1000,
-            )
-            .await;
-            // We had only 1 version per key, we did not prune anything
-            assert_eq!(0, total_pruned.unwrap());
-        }
-        Ok(())
-    }
-
-    #[cfg(not(target_env = "msvc"))]
-    #[tokio::test]
-    async fn test_db_size_after_compaction() -> Result<(), anyhow::Error> {
-        let primary_path = tempfile::tempdir()?.into_path();
-        let perpetual_db = Arc::new(AuthorityPerpetualTables::open(&primary_path, None));
-        let total_unique_object_ids = 100_000;
-        let num_versions_per_object = 10;
-        let ids = ObjectID::in_range(ObjectID::ZERO, total_unique_object_ids)?;
-        for id in ids {
-            for i in (0..num_versions_per_object).rev() {
-                perpetual_db.objects.insert(
-                    &ObjectKey(id, SequenceNumber::from(i)),
-                    &Object::immutable_with_id_for_testing(id),
-                )?;
-            }
-        }
-        perpetual_db.objects.rocksdb.flush()?;
-        let before_compaction_size = get_size(primary_path.clone()).unwrap();
-
-        let total_pruned = AuthorityStorePruner::prune_objects(2, &perpetual_db.objects);
-        info!("Total pruned keys = {:?}", total_pruned);
-        let start = ObjectKey(ObjectID::ZERO, SequenceNumber::MIN);
-        let end = ObjectKey(ObjectID::MAX, SequenceNumber::MAX);
-        perpetual_db.objects.compact_range(&start, &end)?;
-
-        let after_compaction_size = get_size(primary_path).unwrap();
-
-        info!(
-            "Before compaction disk size = {:?}, after compaction disk size = {:?}",
-            before_compaction_size, after_compaction_size
-        );
-        ma::assert_le!(after_compaction_size, before_compaction_size);
-        Ok(())
-    }
-
-    #[cfg(not(target_env = "msvc"))]
-    #[tokio::test]
-    async fn ensure_tombstone_fragmentation_in_stack_frame_without_ignore_range_delete(
-    ) -> Result<(), anyhow::Error> {
```

Comment:
Some of these tests are relevant as long as we have `delete_range` and ignore_delete_range in the code path